### PR TITLE
[opt](Nereids) support no-key hint parameter

### PR DESCRIPTION
**Pull request by morrySnow**
support hint use parameter without key, like:

```sql
SELECT /*+ query_timeout(3000) */ * FROM t;
```

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

File: regression-test/suites/nereids_syntax_p0/hint.groovy
Comment (gavinchou) [inline regression-test/suites/nereids_syntax_p0/hint.groovy:1]: Is it better to add some UT too?

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Comment by morrySnow** [comment]
run buildall

**Review by starocean999 — APPROVED** [review]

**Review by LiBinfeng-01 — APPROVED** [review]

**Review by yiguolei — APPROVED** [review]

**Pull request merged**